core/metrics-core-api/src/main/java/org/hawkular/metrics/core/api/MetricRequest.java

```diff
@@ -46,19 +40,19 @@
  * @author John Sanda
  */
 @ApiModel(description = "A gauge metric with one or more data points")
-public class Gauge {
+public class MetricRequest<S, T extends DataPoint<S>> {
```

Comment:
Could we keep that simple and declare `MetricRequest<T>`, and below `private final List<DataPoint<T>> data` ?